pymeasure/instruments/ipgphotonics/yar.py

```diff
@@ -98,11 +98,9 @@ def read(self):
         else:
             return reply[-1].strip()
 
-    def check_errors(self):
-        """Empty the read buffer."""
-        got = self.read().split(":")
-        if got[0] == "ERR":
-            raise ConnectionError(f"Connection error {got[-1]}.")
```

Comment:
Where do you deal with the error case now?

Reply:
`read` does already the error checking.
These lines were a superfluous repetition.

Reply:
Btw., tests ensure that the error is still raised.

Reply:
Thanks!